docs/source/events/threading.rst

```diff
@@ -0,0 +1,551 @@
+.. _multithreading-in-napari:
+
+Multithreading in napari
+========================
+
+As described in :ref:`intro-to-event-loop`, ``napari``, like most GUI
+applications, runs in an event loop that is continually receiving and
+responding to events like button presses and mouse events.  This works fine
+until one of the events takes a very long time to process.  A long-running
+function (such as training a machine learning model or running a complicated
+analysis routine) may "block" the event loop in the main thread, leading to a
+completely unresponsive viewer.  The example used there was:
+
+.. code-block:: python
+
+   import napari
+   import numpy as np
+
+
+   with napari.gui_qt():
+       viewer = napari.Viewer()
+       # everything is fine so far... but if we trigger a long computation
+       image = np.random.rand(1024, 512, 512).mean(0)
+       viewer.add_image(image)
+       # the entire interface freezes!
+
+In order to avoid freezing the viewer during a long-running blocking function,
+you must run your function in another thread or process.
+
+Processes, Threads, and ``asyncio``
+-----------------------------------
+
+There are multiple ways to achieve "concurrency" (multiple things happening at
+the same time) in python, each with their own advantages and disadvantages.
+It's a rich, complicated topic, and a full treatment is well beyond the scope
+of this document, but strategies generally fall into one of three camps:
+
+1. Multithreading
+2. Multprocessing
+3. Single-thread concurrency with `asyncio
+   <https://docs.python.org/3/library/asyncio.html>`_
+
+For a good high level overview, see `this post
```

Comment:
fwiw, the only document that ever made sense to me about async is the [trio tutorial](https://trio.readthedocs.io/en/stable/tutorial.html)

Reply:
want me to use that instead?

Reply:
in addition to. =)